src/writer/libtest.rs

```diff
@@ -0,0 +1,807 @@
+// Copyright (c) 2018-2022  Brendan Molloy <brendan@bbqsrc.net>,
+//                          Ilya Solovyiov <ilya.solovyiov@gmail.com>,
+//                          Kai Ren <tyranron@gmail.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! [`libtest`][1] compatible [`Writer`] implementation.
+//!
+//! [1]: https://doc.rust-lang.org/rustc/tests/index.html
+
+use std::{
+    fmt::Debug,
+    io, iter, mem,
+    str::FromStr,
+    time::{Duration, SystemTime},
+};
+
+use async_trait::async_trait;
+use derive_more::From;
+use either::Either;
+use itertools::Itertools as _;
+use serde::Serialize;
+
+use crate::{
+    cli, event, parser,
+    writer::{
+        self, basic::coerce_error, out::WriteStrExt as _, Arbitrary, Normalize,
+        Summarize,
+    },
+    Event, World, Writer, WriterExt as _,
+};
+
+/// CLI options of a [`Libtest`] [`Writer`].
+#[derive(Debug, Clone, clap::Args)]
+pub struct Cli {
+    /// Configure formatting of output.
+    #[clap(long, name = "json")]
+    pub format: Option<Format>,
```

Comment:
@ilslv but why `name = "json"` when `--format=json` is used in CLI? 🤔

Reply:
@tyranron `long` and `short` actually describe the argument's "name", while name is used to describe a provided value:

[Example][1]:
```rust
    /// Regex to filter scenarios by their name.
    #[clap(
        short = 'n',
        long = "name",
        name = "regex",
        visible_alias = "scenario-name",
        global = true
    )]
    pub re_filter: Option<Regex>,
```
Output:
```
    -n, --name <regex>
            Regex to filter scenarios by their name

            [aliases: scenario-name]
```


[1]: https://github.com/cucumber-rs/cucumber/blob/41812edf21a539a66b9a30fecd10396414b1b367/src/cli.rs#L95-L102